File: tool/src/org/antlr/v4/parse/ResyncToEndOfRuleBlock.java
/*
 * Copyright (c) 2012 The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD-3-Clause license that
 * can be found in the LICENSE.txt file in the project root.
 */

package org.antlr.v4.parse;

/** Used to throw us out of deeply nested element back to end of a rule's
 *  alt list. Note it's not under RecognitionException.
 */
public class ResyncToEndOfRuleBlock extends RuntimeException {
	private static final long serialVersionUID = 6104510295480069276L;
}

File: runtime/CSharp/Antlr4.Tool/Parse/ResyncToEndOfRuleBlock.cs
// Copyright (c) Terence Parr, Sam Harwell. All Rights Reserved.
// Licensed under the BSD License. See LICENSE.txt in the project root for license information.

namespace Antlr4.Parse
{
    using Exception = System.Exception;

    /** Used to throw us out of deeply nested element back to end of a rule's
     *  alt list. Note it's not under RecognitionException.
     */
    public class ResyncToEndOfRuleBlock : Exception
    {
        public ResyncToEndOfRuleBlock()
        {
        }

        public ResyncToEndOfRuleBlock(string message)
            : base(message)
        {
        }

        public ResyncToEndOfRuleBlock(string message, Exception innerException)
            : base(message, innerException)
        {
        }
    }
}
